berkdb/db/db_truncate.c

```diff
@@ -180,15 +181,14 @@ __db_cursor_check(dbp)
 	for (found = 0, ldbp = __dblist_get(dbenv, dbp->adj_fileid);
 	    ldbp != NULL && ldbp->adj_fileid == dbp->adj_fileid;
 	    ldbp = LIST_NEXT(ldbp, dblistlinks)) {
-		MUTEX_THREAD_LOCK(dbenv, dbp->mutexp);
-		for (dbc = TAILQ_FIRST(&ldbp->active_queue);
+		for (dbc = __db_lock_aq(dbp, ldbp, &cq);
```

Comment:
This only returns cursors held by this thread, no?  IIRC, this was pretty slow- we replaced this with fastinit a while back.  I don't know if it's getting called anymore.

Reply:
Yes. The code is supposed to loop over all cursors held against the dbp, not just the ones held by the thread. You're right that the code isn't getting called anymore. I changed it for completeness.